singlestoredb/management/job.py

```diff
@@ -5,20 +5,26 @@
 from typing import Dict
 from typing import List
 from typing import Optional
-from enum import Enum
+from enum import StrEnum
 
 from .utils import to_datetime
 from .manager import Manager
 
 
-
-class TargetType(Enum):
+class TargetType(StrEnum):
     WORKSPACE = "Workspace"
     CLUSTER = "Cluster"
     VIRTUAL_WORKSPACE = "VirtualWorkspace"
 
+    @classmethod
+    def from_str(cls, s):
+        try:
+            return cls[s.upper()]
```

Comment:
This will need to be able to add underscores as well to handle the camel-case to all-caps. There is a function in `.utils` named `camel_to_snake` that can be used. After importing that, you can do `cls[camel_to_snake(s).upper()]`. You can do the same thing for `Status` just in case a camel-case value is ever added.